src/components/HelpScoutBeacon/BeaconHeadScripts.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Helmet } from 'react-helmet'
+import { GU } from '../../utils'
+
+const BEACON_EMBED =
+  '!function(e,t,n){function a(){var e=t.getElementsByTagName("script")[0],n=t.createElement("script");n.type="text/javascript",n.async=!0,n.src="https://beacon-v2.helpscout.net",e.parentNode.insertBefore(n,e)}if(e.Beacon=n=function(t,n,a){e.Beacon.readyQueue.push({method:t,options:n,data:a})},n.readyQueue=[],"complete"===t.readyState)return a();e.attachEvent?e.attachEvent("onload",a):e.addEventListener("load",a,!1)}(window,document,window.Beacon||function(){});'
+const HELPSCOUT_ID = "'163e0284-762b-4e2d-b3b3-70a73a7e6c9f'"
+const BEACON_INIT = "window.Beacon('init'," + HELPSCOUT_ID + ')'
+
+const BeaconHeadScripts = ({ optedIn }) => {
+  if (!optedIn) {
+    return null
+  }
+
+  return (
+    <Helmet>
```

Comment:
What would you think about trying to skip `react-helmet` and the inline scripts?

```js
// We could use styled-components’ createGlobalStyle for CSS
const HelpscoutStyle = createGlobalStyle`
  /* … */
`

// And this could replace the two inline scripts
function initBeacon(helpscoutId) {
  if (window.Beacon) {
    return window.Beacon
  }

  window.Beacon = (method, options, data) => {
    if (window.Beacon.readyQueue) {
      window.Beacon.readyQueue.push({ method, options, data })
    }
  }
  window.Beacon.readyQueue = [{ method: 'init', options: helpscoutId }]

  const insertScript = () => {
    const script = document.createElement('script')
    script.async = true
    script.src = 'https://beacon-v2.helpscout.net'
    document.body.appendChild(script)
  }

  if (document.readyState === 'complete') {
    insertScript()
  } else {
    window.addEventListener('load', insertScript)
  }

  return window.Beacon
}

function useHelpScoutBeacon(optedIn) {
  const [beacon, setBeacon] = useState(null)

  useEffect(() => {
    if (optedIn && !beacon) {
      setBeacon(() => initBeacon(HELPSCOUT_ID))
    }

    // Here we could destroy it, but I’m not sure if HelpScout’s
    // destroy / init actions are fully sync, so better leave it always on once it’s here.
    // return () => if (beacon) beacon('destroy') 
  }, [optedIn, beacon])

  return beacon
}

// it might not make sense at this point to keep this component :D 
const BeaconHeadScripts = ({ optedIn }) => {
  useHelpScoutBeacon(optedIn)
  return optedIn ? <HelpscoutBeaconStyle /> : null
}
```

Reply:
Your suggestion is a specific implementation for `HelpScout`, I avoided doing so for readibility and having to maintain that part of the codebase into the future.
Keeping the inline scripts makes it easier to update if there's any need to do so and we could use the same mechanism for something else we might want to inject instead of having to do independent components for each one.

Reply:
> Your suggestion is a specific implementation for HelpScout, I avoided doing so for readibility and having to maintain that part of the codebase into the future.

It’s doing what their inline script is doing, so I’m not too worried about that. The important thing it gives us is control over it, so we can know exactly when it is ready. But it’s true that it might be a problem in the future if they ask their customers to update their init script, so I let you decide! One thing we could still do to avoid using `react-helmet` (to avoid [increasing the bundle size](https://bundlephobia.com/result?p=react-helmet@5.2.1) for something that we can do with little code) would be to inject the scripts ourselves:

```jsx

// /!\ Untested code

function useHelpScoutBeacon(optedIn) {
  const [beacon, setBeacon] = useState(null)

  // Load the script if it doesn’t exist yet and optedIn
  // is true, then set the value of Beacon.
  useEffect(() => {
    if (optedIn && !beacon) {
      const scripts = [BEACON_EMBED, BEACON_INIT].map(source => {
        const script = document.createElement('script')
        script.innerHTML = source
        return document.body.appendChild(script)
      })
      window.Beacon('init', HELPSCOUT_ID)
      setBeacon(window.Beacon)
    }
    return () => embedScript.remove()
  }, [optedIn, beacon])

  return beacon
}
```

Reply:
> Until then I consider this to be an exception to the rule, but running code that we don’t control relying on a remote HTTP server is something we should avoid as much as possible.

Agree 100%, but for a feature that I understand to be "let them have it and see if they use it", let's not worry about being too optimized for now :).

Reply:
Yes totally, I should have mentioned here that the original intention was to avoid using `react-helmet` by using `createGlobalStyle` and `document.createElement`, but the other one is to explicitly re-render when `Beacon` is initialized: https://github.com/aragon/aragon/pull/758#discussion_r282798033

Reply:
If we end up removing this feature then removing `react-helmet` will be easy, I originally tried with an implementation similar to your last comment @bpierre and could not get it to work.
I understand your desire to remove it, is it that strong an opinion?

Reply:
Let’s keep it for now, but it’s definitely a a module we should try to eliminate IMO: it’s easy to replace with a simple `document.createElement("script")`, it has several nested dependencies, and it adds [6.4k (gzipped)](https://bundlephobia.com/result?p=react-helmet@5.2.1) to the app.